Приложение "Заметки"

GrafScore opened the pull request.

Commit:
Написал код на языке котлин, согслано заданию.
Произвел проверку, недочетов не обнаружил.

Commit:
Написал код на языке котлин, согслано заданию.
Произвел проверку, недочетов не обнаружил.

Charmandik commented on .idea/misc.xml:
Попробуй добавить этот тип файлов в гитигнор, их не обязательно отслеживать в гите, они генерируемые и у каждого свои

Charmandik commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,136 @@
-    println("Hello World!")
+import java.util.Scanner
+
+class Note(val text: String)
```
старайся разносить разные классы или интерфейсы по разным файлам. Один класс/интерфейс в один файл. Это позволяет гораздо легче ориентироваться в структуре проекта и упрощает чтение, так как ты заходишь в нужный тебе класс и читаешь только то, чем занимается класс

Charmandik commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,136 @@
+        println("Введите название нового архива (или введите 2 для отмены):")
+        val archiveName = scanner.nextLine()
+        when {
+            archiveName == "2" -> {
```
Числа, которые просто так вливаются в код, зачастую очень сложно понять и их называют магическими числами. Используй константы, для того, чтобы дать имена своим числам Рекомендую к прочтению статью на эту тему : https://bureau.ru/soviet/20230406/

Charmandik commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,136 @@
+        println(note.text)
+    }
+
+    @JvmStatic
```
тебе не нужно добавлять эту аннотацию, так как она нужна только для случаев, когда вызываешь Kotlin код из Java кода

Commit:
Добавил возможность добавлять содержимое заметки и просматривать соддержимое заметки.
Класс Note перенес в отдельный файл.
Исправил ситуацию с номером выхода из "Заметок" и "Архива". Теперь это последняя цифра в списке.

Commit:
Добавил возможность добавлять содержимое заметки и просматривать соддержимое заметки.
Класс Note перенес в отдельный файл.
Исправил ситуацию с номером выхода из "Заметок" и "Архива". Теперь это последняя цифра в списке.
